Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import TensorFlow
+
+/// A BetweenFactor alternative that uses the Chordal (Frobenious) norm on rotation for Rot3
+/// Please refer to Carlone15icra (Initialization Techniques for 3D SLAM: a Survey on Rotation Estimation and its Use in Pose Graph Optimization)
+/// for explanation.
+public struct FrobeniusFactorRot3: LinearizableFactor
+{
+  public typealias Variables = Tuple2<Vector9, Vector9>
```

Comment:
I don't understand why this is Vector9. Frobenius error is defined on Rot3, right? Refer to GTSAM. I would contend: it should do exactly the same as GTSAM and be defined on any type that has a differentiable "vec" method.

Reply:
I think it is not, because in chordal initialization we actually wanted to offset from the SO(3) manifold (relaxed chordal) and then map back to SO(3) with `Rot3.ClosestTo`.

Reply:
Unless what you want is not what the `InitializePose3` does in GTSAM?

Reply:
Ah, yes, I think you are right. But, then it would still be Matrix3 not Vector9. And, BTW, this is then different from the FrobeniusFactor in GTSAM?? Talk about creating confusion :-/ I believe that if you use the same name as in GTSAM it should have the same functionality, and *also* that if we find improvements here, they should make it back to GTSAM. In particular, maybe FrobeniusFactor in GTSAM, being defined for any type that has “vec”, can be used in chordal in it in GTSAM to make it much simpler? Nothing says that the type used for FrobeniusFactor cannot be “relaxed” in how it updates.

Trying to give enough feedback here to get you to think bigger picture, in this case spanning two libraries, and “one stone,...”

Reply:
Sure :) I renamed the factor to `RelaxedRotationFactorRot3`, since it is still not possible to make it generic yet.

On the GTSAM part, yes I think it is definitely possible to reuse FrobeniusFactor somehow, by similarly creating a rotation-only NLFG, linearize and solve, but I think it will be slower than the current one as the current one creates a GFG directly.